_high_availability/003-Cluster-Configuration.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: page
+title:  "Cluster configuration"
+date:   2015-01-27 22:02:36
+categories:
+permalink: /ha/cluster.html
+---
+
+
+The cluster
+------------------
+
+Now that we have a correctly replicating database, we need to establish a mechanism for managing the actual failover and promotion of nodes. For this, we will create a cluster
+using pacemaker and corosync. 
+
+Packages
+==============
+
+The following steps must be run in all nodes of our DB cluster.
+
+Let's start by installing the appropriate packages
+
+    $ sudo apt-get install corosync pacemaker pacemaker-cli-utils cluster-glue
+
+Then we need to update the postgres RA (resource agent) as the one in the standard distribution is a bit old
+
+    $ sudo wget https://raw.githubusercontent.com/ClusterLabs/resource-agents/master/heartbeat/pgsql -O /usr/lib/ocf/resource.d/heartbeat/pgsql
+
+
+Corosync configuration
```

Comment:
They apply the proper vips to nodes (master and slave nodes) and promote one server to master. The newly introduced slave (after failover) will assume the slave serving IP.